core/modules/user/user.admin.inc

```diff
@@ -168,6 +168,49 @@ function user_admin_settings($form, &$form_state) {
     }
   }
 
+  // Password settings.
+  $form['password_constraints'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Password constraints'),
+    '#weight' => -1,
+  );
+  $form['password_constraints']['user_password_enforce_strong'] = array(
+    '#type' => 'radios',
+    '#title' => t('Password constraints behavior'),
+    '#options' => array(
+      0 => t('Only indicate password strength'),
+      1 => t('Enforce strong passwords'),
+    ),
+    '#default_value' => $config->get('user_password_enforce_strong'),
+  );
+  $constraint_defaults = $config->get('user_password_constraints');
+  $form['password_constraints']['user_password_constraints'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('Set constraints'),
+    '#options' => array(
+      'not_same_as_username' => t('Password not same as username'),
+      'not_same_as_email' => t('Password not same as account e-mail'),
+      'case_insensitive_comparison' => t('Ignore case when comparing'),
+      'min_length_enforce' => t('Set a minimum password length'),
+    ),
+    '#default_value' => $constraint_defaults,
+  );
+  $form['password_constraints']['user_password_min_length'] = array(
+    '#type' => 'number',
+    '#title' => t('Minimum length'),
+    '#min' => 1,
+    '#max' => 64,
+    '#default_value' => $config->get('user_password_min_length'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="user_password_constraints[min_length_enforce]"]' => array('checked' => TRUE),
+      ),
+      'required' => array(
```

Comment:
@herbdool there's a known bug with `#states` and `visible`: the red asterisk is added to the field, but the validation never actually kicks in. See https://github.com/backdrop/backdrop-issues/issues/2732

So if we really need this to be required (we should), then we need to implement our own validation for it in a `user_admin_settings_validate()`.

Reply:
...see my (nasty) workaround here: https://github.com/backdrop/backdrop/pull/2783/files#diff-805bfa8b35104aa06009fc84f21666a5R117

Reply:
I think we don't really need to make it required. The simplest approach is to get rid of it for now.